code/statsd/src/main/scala/com/avast/datadog4s/statsd/StatsDMetricFactory.scala

```diff
@@ -16,7 +16,7 @@ class StatsDMetricFactory[F[_]: Sync](
   defaultTags: collection.immutable.Seq[Tag]
 ) extends MetricFactory[F] {
 
-  private[this] val clock = Clock.create[F]
+  private[this] val clock: Clock[F] = Clock[F]
```

Comment:
So, do we need this field at all?

Reply:
it's not strictly needed but it's useful to have it explicit (as opposed to implicit) ... otherwise there is issues in testing because `Sync[F] extends Clock[F]` and compiler gets confused and it's very hard to inject Mock clock into tests :( sorry